### DISCUSSION TITLE
Question about OptionAsync

### DISCUSSION THREAD

**Discussion by JonTheTurnip** [louthy/language-ext#1207] [Q&A] [answered]
Hi all,

I'm just starting out on my functional journey, I'm in awe of this library!  I hope to properly understand it all one day.

I have a question about using OptionAsync.  I think the example below illustrates what I'm trying to do.  

I have an async method that I want to Bind and therefore I'm looking at OptionAsync rather than Option.  The problem is that I'm getting the error below, what's the correct way to handle this?

Many thanks :)

![image](https://user-images.githubusercontent.com/10959562/224778604-7dcf07ef-657c-4b2c-8df5-1986cb17d0cd.png)

**Comment by evermanwa** [comment, marked as answer]
Try to use "return func().ToAsync();" instead. When a task is wrapping a monad, the ToAsync will convert them to the async equivalent. So Task<Either<A,B>> will be converted to EitherAsync<A,B> if you call ToAsync on it.

**Reply by JonTheTurnip** [reply]
Great, thanks so much for the reply.
